kubespawner/spawner.py

```diff
@@ -1998,6 +1998,14 @@ def _load_profile(self, profile_name):
     # used for warning about ignoring unrecognised options
     _user_option_keys = {'profile',}
 
+    def _init_profile_list(self, profile_list):
+        # generate missing slug fields from display_name
+        for i, v in enumerate(profile_list):
+            if 'slug' not in profile_list[i]:
+                profile_list[i]['slug'] = slugify(profile_list[i]['display_name'])
```

Comment:
Since we are modifying the items, we don't need the index:
```suggestion
        for profile in profile_list:
            if 'slug' not in profile:
                profile['slug'] = slugify(profile['display_name'])
```